src/main/scala/lantern/dslapi.scala

```diff
@@ -69,8 +69,23 @@ with CastingOps {
   implicit def repStrToSeqOps(a: Rep[String]) = new SeqOpsCls(a.asInstanceOf[Rep[Seq[Char]]])
   implicit class BooleanOps2(lhs: Rep[Boolean]) {
     def &&(rhs: =>Rep[Boolean])(implicit pos: SourceContext) =
-    __ifThenElse(lhs, rhs, unit(false)) }
-//  override def boolean_and(lhs: Rep[Boolean], rhs: Rep[Boolean])(implicit pos: SourceContext): Rep[Boolean] = __ifThenElse(lhs, rhs, unit(false))
+      __ifThenElse(lhs, rhs, unit(false))
+  }
+  // override def boolean_and(lhs: Rep[Boolean], rhs: Rep[Boolean])(implicit pos: SourceContext): Rep[Boolean] = __ifThenElse(lhs, rhs, unit(false))
+
+  implicit def repArrayToTensorOps[T: Manifest](a: Rep[Array[T]]) = new TensorOpsCls(a)
```

Comment:
I'm concerned that `repArrayToTensorOps` overpowers [`repArrayToArrayOps`](https://github.com/TiarkRompf/virtualization-lms-core/blob/88b4ca941d747b49b25c491acde2d56a8e8600c6/src/common/ArrayOps.scala#L26) during implicit resolution.
If so, I think that means all `ArrayApply` and `ArrayUpdate` ops are rewritten to `TensorApply` and `TensorUpdate`.

To avoid this, it seems necessary to distinguish `Rep[Array[Float]]` allocated by `TensorNew` from other `Rep[Array[Float]]`. Is this possible/desirable (perhaps via type annotation)? @TiarkRompf